### ISSUE TITLE
Get rid of Iron Router

### ISSUE THREAD

**Issue by johanbrook** [lookback/meteor-emails#45]
It doesn't fulfill any purpose besides routing. 

We can use `WebApp.connectHandlers` for that.

**Comment by mccormjt** [comment]
Also this can cause some unexpected side effects. Right now I cant use the preview feature because I have some IronRouter global hooks that interfere with these routes :(

**Comment by johanbrook** [comment]
That's not Iron Router though – that's how routes work in general. You can configure the emails prefix to be something else of you have a `/emails/preview` route already.

**Comment by satyavh** [comment]
You can't use FlowRouter for this, it's client only. You have to use something like Picker.
But I also vote for 2nd option, the less dependencies the better. 

I can probably do this because we're just moving our app from IR to FR , which means we need to get rid of the IR dependency before we can use this package again.

**Comment by johanbrook** [comment]
> You can't use FlowRouter for this, it's client only.

Ah, true. Forgot that. Would love a PR for this.

**Comment by satyavh** [comment]
So I implemented this with `meteorhacks:picker`, it works great.

I started off with WebApp but actually there's a lot of stuff you need to do to process the routes and params if you want to keep the package flexible. Besides WebApp docs are kind of lacking. Picker is just WebApp plus just enough to make live easy, so I took the easy road (Picker is really small, check the code).

I still need to replace the `emailForUrl` helper, which also depends on IR.

Once I'm done I'll do a PR (unless you really want to go for a WebApp solution)

**Comment by johanbrook** [comment]
Thank you!

Legit thoughts on WebApp vs. <other router>. The things I've done with `WebApp.connectHandlers` has ended up with me wishing I'd gone with a router package instead.

Feel free to file a PR and continue pushing commits to it.

**Comment by satyavh** [comment]
Is that `emailUrlFor` helpers actually going to work without IR? Because if I understand correctly, you can just pass any IR route name in there and then it creates an absolute url for your email. But without IR there are no IR route names, and you cannot lookup a route.

**Comment by satyavh** [comment]
Or we can just leave it as is, and if you use IR you can use this helper. And then we can add some functionality for FR in that helper. What do you think?

(hmmm FR has a bug looking up route path by route name...)

**Comment by johanbrook** [comment]
@satyavh 

> Is that emailUrlFor helpers actually going to work without IR?

Nope, they won't. But we can have Iron Router as a weak dependency, and check if it's included in the main app:

``` coffeescript
if Package['iron:router']
  # do stuff
```

**Comment by elie222** [comment]
The iron router integration can mess things up. I have a global hook that checks if a user is logged in or not, but when I go to the preview page, I get an error saying:

```
Error: Meteor.userId can only be invoked in method calls. Use this.userId in publish functions.
W20151219-23:58:51.404(2)? (STDERR)     at AccountsServer.userId (accounts_server.js:80:13)
W20151219-23:58:51.404(2)? (STDERR)     at AccountsServer.user (accounts_common.js:53:23)
W20151219-23:58:51.404(2)? (STDERR)     at Object.Meteor.user (accounts_common.js:230:19)
W20151219-23:58:51.404(2)? (STDERR)     at [object Object].mustBeSignedIn (router.js:34:15)
```

**Comment by elie222** [comment]
I solved my issue my wrapping the revelant global hooks in a `Meteor.isClient` block.

**Comment by johanbrook** [comment]
Fixed in https://github.com/lookback/meteor-emails/pull/47 and will be published in 0.7.0.
